src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotUseStackallocInLoopsTests.cs

```diff
@@ -0,0 +1,174 @@
+{
+    public class DoNotUseStackallocInLoopsTests
+    {
+        [Fact]
+        public async Task NoDiagnostics_StackallocNotInLoop()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                unsafe class TestClass {
+                    private static void NoLoop() {
+                        byte* ptr1 = stackalloc byte[1];
+                        Span<byte> ptr2 = stackalloc byte[2];
+
+                        Label1:
+                        Span<byte> ptr3 = stackalloc byte[3];
+                        goto Label1; // false negative, but too difficult to track well with few false positives
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task NoDiagnostics_StackallocInLoopWithBreak()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                unsafe class TestClass {
+                    private static void ForLoop() {
+                        for (int i = 0; i < 10; i++)
+                        {
+                            if (i == 5)
+                            {
+                                byte* ptr = stackalloc byte[1024];
+                            }
+                            break;
+                        }
+                    }
+
+                    private static void WhileLoop() {
+                        while (true)
+                        {
+                            byte* ptr = stackalloc byte[1024];
+                            break;
+                        }
+                    }
+
+                    private static void DoWhile() {
+                        do
+                        {
+                            byte* ptr = stackalloc byte[1024];
+                            return;
+                        }
+                        while (true);
+                    }
+                }");
+        }
+
+        [Fact]
+        public async Task Diagnostics_LoopsWithStackallocPtr()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+                using System;
+                unsafe class TestClass {
+                    private static void ForLoop() {
+                        for (int i = 0; i < 10; i++)
+                        {
+                            byte* ptr = stackalloc byte[1024];
+                        }
+                    }
+
+                    private static void WhileLoop() {
+                        while (true)
+                        {
+                            byte* ptr = stackalloc byte[1024];
+                        }
+                    }
+
+                    private static void DoWhile() {
+                        do
+                        {
+                            byte* ptr = stackalloc byte[1024];
+                        }
+                        while (true);
+                    }
+                }",
+                VerifyCS.Diagnostic().WithSpan(7, 41, 7, 62),
```

Comment:
For future reference, you can also directly add diagnostic markup inline in source, such as say `{|CA2014:stackalloc byte[1024]|}`

Reply:
Ok, I'll change these.